include/swift/AST/Attr.def

```diff
@@ -624,6 +624,11 @@ SIMPLE_DECL_ATTR(reasync, AtReasync,
   ABIBreakingToAdd | ABIBreakingToRemove | APIBreakingToAdd | APIBreakingToRemove,
   110)
 
+SIMPLE_DECL_ATTR(_rethrowsUnchecked, RethrowsUnchecked,
```

Comment:
It would be valuable to land this attribute _before_ the checker changes so that code that relies on the current behaviour (such as `DispatchQueue`) has an opportunity to migrate.